ddsrouter_core/src/cpp/communication/rpc/RPCBridge.cpp

```diff
@@ -365,6 +373,9 @@ void RPCBridge::transmit_(
                     continue;
                 }
 
+                // Perform write + add entry to registry atomically -> avoid reply processed before entry added to registry
+                std::lock_guard<std::recursive_mutex> lock(request_writers_[service_registry.first]->get_mutex());
```

Comment:
There is a Race Condition that I do not know it is handled in enable of ServiceRegistry, that (maybe, not sure) could lead to not sending first message in transmit_::request_topic because service registry is not enabled

Reply:
This is an intentional race condition; `discovered_service` might be taking `on_transmission_mutex` to avoid the race condition, but we might miss a server discovery (and registry activation) in the middle of a transmission.